src/Bicep.Core.IntegrationTests/PrettyPrint/PrettyPrinterTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+using Bicep.Core.Parser;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.PrettyPrint.Options;
+using Bicep.Core.Samples;
+using Bicep.Core.Syntax;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.IntegrationTests.PrettyPrint
+{
+    [TestClass]
+    public class PrettyPrinterTests

```

Comment:
>PrettyPrinterTests [](start = 17, length = 18)

Although it's been rare lately, we sometimes have to add new Visit* methods to `SyntaxVisitor`. If we do that, I'm guessing the pretty printer would drop some AST nodes? Would any of the tests catch it? This might be tricky though because we can't rely on any of the existing assert files because in the worst case, all the spans of all the nodes get adjusted. THere might be better ideas, but one option could be just to count nodes by type between unformatted ast and formatted ast. Counts by node type should match. (I have `SyntaxAggregator` class that might be useful for this.)

Reply:
Anthony brought up the same question. The `PrintProgram_ProgramWithoutDiagnostics_ShouldRoundTrip` bellow will catch that. It basically re-parses the formatted program and checking if the new AST is identical to the original one, by printing them to strings with newlines and whitespaces removed and comparing the strings.

Reply:
Perfect!